src/main/java/com/microsoft/jenkins/containeragents/aci/AciService.java

```diff
@@ -65,6 +72,14 @@ public static void createDeployment(final AciCloud cloud,
             ObjectNode.class.cast(tmp.get("variables")).put("jenkinsInstance",
                     Jenkins.getInstance().getLegacyInstanceId());
 
+            if (logAnalyticsCredentials != null) {
+                ObjectNode.class.cast(tmp.get("variables")).put("workspaceId",
+                        logAnalyticsCredentials.getUsername());
+
+                ObjectNode.class.cast(tmp.get("variables")).put("workspaceKey",
+                        logAnalyticsCredentials.getPassword().getPlainText());
```

Comment:
credentials should be a parameter marked with a type secureString

https://github.com/jenkinsci/azure-vm-agents-plugin/blob/75f3235b72021daa6ae0e5fd415d3c7652b394bd/src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java#L557-L558

🚫 blocker